OpenDreamRuntime/DreamManager.cs

```diff
@@ -171,8 +171,20 @@ internal partial class DreamManager : IDreamManager {
                         // i dont believe this will **ever** be called, but just to be sure, funky errors /might/ appear in the future if someone does a fucky wucky and calls this on a deleted object.
                         throw new Exception("Cannot create reference ID for an object that is deleted");
                     }
-
-                    refType = RefType.DreamObject;
+                    switch(refObject){
+                        case DreamObjectTurf: refType = RefType.DreamObjectTurf; break;
+                        case DreamObjectMob: refType = RefType.DreamObjectMob; break;
+                        case DreamObjectArea: refType = RefType.DreamObjectArea; break;
+                        case DreamObjectClient: refType = RefType.DreamObjectArea; break;
+                        case DreamObjectImage: refType = RefType.DreamObjectImage; break;
+                        case DreamList: refType = RefType.DreamObjectList; break;
+                        default: {
+                            refType = RefType.DreamObjectDatum;
+                            if(refObject.IsSubtypeOf(_objectTree.Obj))
+                                refType = RefType.DreamObject;
```

Comment:
Ref ID 0xF should be reserved for DreamList but not any of its descendants.
```suggestion
                        default: {
                            refType = RefType.DreamObjectDatum;
                            if(refObject.IsSubtypeOf(_objectTree.Obj))
                                refType = RefType.DreamObject;
                            else if (refObject.GetType() == typeof(DreamList))
                                refType = RefType.DreamObjectList;
```